sdk/storage/storage-queue/RecordAndPlayback.md

```diff
@@ -0,0 +1,199 @@
+
+- Thus, the test recording[`recording_<test-title>.{js|json}`] takes care of `beforeEach` and `afterEach` along with the actual test inside the `it`-block. Recordings corresponding to `beforeEach` or `afterEach` sections are saved along with the test recordings.
+
+- Follow the below template for adding any new test. `before` and `after` sections are optional and can be added whenever necessary, `beforeEach` and `afterEach` sections are compulsory.
+
+  ```typescript
+  import { record } from "./utils/recorder";
+
+  describe("<describe-block-title>", () => {
+    // before section is optional
+    before(async function() {
+      recorder = record(this);
+      /*Place your code here*/
+      recorder.stop();
+    });
+
+    // after section is optional
+    after(async function() {
+      recorder = record(this);
+      /*Place your code here*/
+      recorder.stop();
+    });
+
+    beforeEach(async function() {
+      recorder = record(this);
+      /*Place your code here*/
+    });
+
+    afterEach(async () => {
+      /*Place your code here*/
+      recorder.stop();
+    });
+
+    it("<test-title>", async () => {
+      /*Place your code here*/
+    });
+  });
+  ```
+
+- **Saving randomly generated information/unique info of a test run. Example - `getUniqueName` in storage-queue**
+
+  - Before Record and Playback, `getUniqueName` was being imported from `utils` in the `.spec.ts` test files. `getUniqueName` generates a new value each time we run. Retaining that unique value is crucial in order to replay the http request during `"playback"` mode, which also implies that any unique information related to the test must be saved in order to replay the http requests.
+
+  - When `recorder.stop();` is called for a test, we save that unique information corresponding to the test run along with the test recording.
+
+  - During `playback` mode, the saved unique information must be pulled out from the existing recording in order to replay the http requests.
+
+    For the case of `getUniqueName`, to keep things clean in the `spec.ts` files, we have moved the `getUniqueName` into the `recorder.ts`.
+    i.e., a wrapper is added around `getUniqueName` in `recorder.ts` which takes care of the `TEST_MODE`
+
+    - if `TEST_MODE` = `"record"`, `recorder.getUniqueName()` will work the same as the usual `getUniqueName()`.
+    - if `TEST_MODE` = `"playback"`, `recorder.getUniqueName()` will pull out the value from existing recording.
+
+  [Same is the case with `newDate` function]
+
+  - Any uniquely generated information of the test run that is important to replay the http request must be saved along with the recordings in the record mode.
+
+- If a new test/test-suite is added, execute the test/test-suite(or all the tests) by setting the env variable `TEST_MODE = record` and **commit** the generated recording files.
+
```

Comment:
What to do if the description of a test or test suite is modified?  It would be nice to cover this case.

Reply:
ah nvm it's covered above.